File: Data Structures & Algorithms/binary-search/submission-2.py
import math
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        first, last = 0, len(nums)

        while first < last:
            mid = first + ((last - first) // 2)
            if nums[mid] >= target:
                last = mid
            elif nums[mid] < target:
                first = mid + 1
        return first if (first < len(nums) and nums[first] == target) else -1
        
        

File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for(int i = 0; i < nums.length; i++) {
            for(int j = i + 1; j < nums.length; i++) {
                if(arr[i] == arr[j]) {
                    return true;
                }
            }
        }
        return false;
    }
}
